src/lib/core/TLVWriter.cpp

```diff
@@ -290,12 +290,26 @@ CHIP_ERROR TLVWriter::PutBytes(Tag tag, const uint8_t * buf, uint32_t len)
 
 CHIP_ERROR TLVWriter::PutString(Tag tag, const char * buf)
 {
-    size_t len = strlen(buf);
+    if (buf == nullptr)
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    else if (mMaxLen == 0)
+        return CHIP_ERROR_INCORRECT_STATE;
+
+    // No length is provided, so we cannot assume a well-formed C string.
```

Comment:
This comment makes no sense.  No length is provided, so you have to have a well-formed C string to have sane behavior.  Maybe we can't assume that, and then we want to limit the scope of the insanity, but if so the comment should say that.

Reply:
I see, yes we must assume some sanity since we cannot prevent crashing on a completely corrupted pointer. Rephrased the comment